xnmt/training_task.py

```diff
@@ -160,12 +160,14 @@ def _augment_data_next_epoch(self):
       if self.training_state.epoch_num > 0:
         logger.info('using reloaded data')
       # reload the data   
+      self.model.src_reader.train = self.model.trg_reader.train = True
```

Comment:
In other components, different train/test behavior is achieved by listening to the ```on_set_train()``` event. It might keep things cleaner (and preserve consistency among training regimens) to use that here also unless there is a reason not to? From taking a quick look I think one would have to add a ```set_train(True)``` trigger to ```advance_epoch()```, but otherwise events are probably already triggered at the proper times.

Reply:
Done